pandas/core/frame.py

```diff
@@ -5899,6 +5899,10 @@ def _construct_result(self, result) -> "DataFrame":
 -----
 Matching NaNs will not appear as a difference.
 
+Can only compare identically-labeled DataFrame objects,
+this means DataFrames with the same shape and
+identical row, column labels.
```

Comment:
if the user tries this they will get a `ValueError: Can only compare identically-labeled DataFrame objects` 

so can you also add a Raises section to the docstring.

Reply:
Yes, makes sense, added.